src/basic/io-util.c

```diff
@@ -125,6 +127,9 @@ int loop_write(int fd, const void *buf, size_t nbytes, bool do_poll) {
                 p = buf;
         }
 
+        if (timeout != 0)
+                end = timeout == USEC_INFINITY ? USEC_INFINITY : usec_add(now(CLOCK_MONOTONIC), timeout);
```

Comment:
i'd probably do it like this:

```c
end = timeout == 0 || timeout == USEC_INFINITY ? timeout : usec_add(now(CLOCK_MONOTONIC, timeout));
```

why? simply to ensure `end` is always initialized to something reasonable, because static checkers are generally pretty bad at correctly determining whether variables are always properly initialized, and might otherwise get confused.